samples/apps/copilot-chat-app/webapi/Skills/ChatHistorySkill.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System.Globalization;
+using System.Text.Json;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using SKWebApi.Skills;
+using SKWebApi.Storage;
+
+namespace SemanticKernel.Service.Skills;
+
+/// <summary>
+/// ChatHistorySkill provides functions to store and retrieve chat information in memory,
+/// as well as functions to extract memories from context.
+/// </summary>
+public class ChatHistorySkill
+{
+    /// <summary>
+    /// Returns the name of the semantic text memory collection that stores chat message information.
+    /// </summary>
+    /// <param name="chatId">Chat ID that is persistent and unique for the chat session.</param>
+    public static string MessageCollectionName(string chatId) => $"{chatId}-messages";
+
+    private readonly ChatMessageRepository _chatMessageRepository;
+    private readonly ChatSessionRepository _chatSessionRepository;
+
+    public ChatHistorySkill(
+        ChatMessageRepository chatMessageRepository,
+        ChatSessionRepository chatSessionRepository)
+    {
+        this._chatMessageRepository = chatMessageRepository;
+        this._chatSessionRepository = chatSessionRepository;
+    }
+
+    /// <summary>
+    /// Create a new chat session in memory.
+    /// </summary>
+    /// <param name="title">The title of the chat.</param>
+    /// <param name="context">Contains the memory, "userId", and "userName".</param>
+    /// <returns>An unique chat ID.</returns>
+    /// <returns>The initial chat message in a serialized json string.</returns>
+    [SKFunction("Create a new chat session in memory.")]
+    [SKFunctionName("CreateChat")]
+    [SKFunctionInput(Description = "The title of the chat.")]
+    [SKFunctionContextParameter(Name = "userId", Description = "Unique and persistent identifier for a user.")]
+    [SKFunctionContextParameter(Name = "userName", Description = "Name of the user.")]
+    public async Task<SKContext> CreateChatAsync(string title, SKContext context)
+    {
+        var userId = context["userId"];
+        var userName = context["userName"];
```

Comment:
What if one of those arguments/variables does not exist in the SK context?

Reply:
This will throw an exception and the front end will show the error message.